src/server/scripts/EasternKingdoms/zone_silverpine_forest.cpp

```diff
@@ -5301,6 +5301,275 @@ struct npc_silverpine_generic_actor_fenris_keep : public ScriptedAI
+                    _events.ScheduleEvent(EVENT_NOWHERE_TO_RUN + 7, 1s + 200ms);
+                    break;
+
+                case EVENT_NOWHERE_TO_RUN + 7:
+                    me->SetFacingTo(1.2391838f);
+                    DoCastSelf(SPELL_EJECT_PASSENGER_01);
+                    if (Player* player = ObjectAccessor::GetPlayer(*me, _playerGUID))
+                        player->GetMotionMaster()->MoveJump(PlayerElemJumpPos, 25.0f, 8.0f);
+                    Talk(TALK_FORTESKI_SAVE_PLAYER);
+                    if (Creature* bloodfang = ObjectAccessor::GetCreature(*me, _bloodfangGUID))
+                        bloodfang->DespawnOrUnsummon(2s);
+                    _events.ScheduleEvent(EVENT_NOWHERE_TO_RUN + 8, 3s + 500ms);
+                    break;
+
+                case EVENT_NOWHERE_TO_RUN + 8:
+                {
+                    if (Player* player = ObjectAccessor::GetPlayer(*me, _playerGUID))
+                    {
+                        player->CompleteQuest(QUEST_NOWHERE_TO_RUN);
+
+                        if (Group* group = player->GetGroup())
+                        {
+                            for (GroupReference* itr = group->GetFirstMember(); itr != nullptr; itr = itr->next())
+                            {
+                                Player* groupMember = itr->GetSource();
+
+                                if (groupMember && groupMember->IsInMap(player) && groupMember->GetQuestStatus(QUEST_NOWHERE_TO_RUN) == QUEST_STATUS_INCOMPLETE && groupMember->GetDistance(me) <= 75.f)
+                                    groupMember->CompleteQuest(QUEST_NOWHERE_TO_RUN);
+                            }
+                        }
+                    }
+
+                    _events.ScheduleEvent(EVENT_NOWHERE_TO_RUN + 9, 5s);
+                    break;
+                }
+
+                case EVENT_NOWHERE_TO_RUN + 9:
+                {
+                    for (ObjectGuid const& forsakenTrooperGUID : _forsakenTroopersGUID)
+                    {
+                        if (Creature* forsakenTrooper = ObjectAccessor::GetCreature(*me, forsakenTrooperGUID))
+                            forsakenTrooper->DespawnOrUnsummon();
+                    }
+
+                    me->DespawnOrUnsummon();
+                    break;
+                }
+
+                default:
+                    break;
+            }
+        }
+    }
+
+    void CheckAndAddForsakenTroopersToFormation()
+    {
+        std::vector<Creature*> forsakenTrooperList;
+        GetCreatureListWithEntryInGrid(forsakenTrooperList, me, NPC_VETERAN_FORSAKEN_TROOPER, 25.0f);
+
+        for (uint32 i = 0; i < forsakenTrooperList.size(); i++)
+        {
+            _forsakenTroopersGUID[i] = forsakenTrooperList[i]->GetGUID();
+
+            if (Creature* forsakenTrooper = ObjectAccessor::GetCreature(*me, _forsakenTroopersGUID[i]))
+            {
+                forsakenTrooper->GetMotionMaster()->MoveFollow(me, float(1.5f * i), float(M_PI));
```

Comment:
no need to use ObjectAccessor here, you have access to the index already, so you can access the related forsakenTrooper by using `forsakenTrooperList[i]`

Reply:
Feedback applied!